sklearn/tree/_export.py

```diff
@@ -943,6 +944,10 @@ def export_text(
         A list of length n_features containing the feature names.
         If None generic names will be used ("feature_0", "feature_1", ...).
 
+    class_names : list of arguments, default=None
+        Names of each of the target classes in ascending numerical order.
+        Only relevant for classification and not supported for multi-output.
```

Comment:
Is it worth mentioning that the number of element in class_names should be the same length as the number of class in the tree?

Reply:
According to what you said above, if we add another case, indeed it's better to precise that the number of classes should be the same length as the number of element in `class_names`.

So the new documentation would be :

```python
"""
    class_names : list of arguments, default=None
        A list of length n_classes containing the class names.
        Names of each of the target classes in ascending numerical order.
        Only relevant for classification and not supported for multi-output.
        If None generic names will be used ("0", "1", ...).
"""
```

Reply:
Looks good to me :)

Reply:
> If None generic names will be used ("0", "1", ...).

This is incorrect. `dt.fit(X, y)` can be called with `y` being non-numerical strings. Internally, it will make some encoding but this is an implementation detail.

Since you are affecting `dt.classes_` for `None`, then then name are not generic but corresponds to the target seen during `fit` by the decision tree.